server/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -402,15 +395,9 @@ public static TermsQueryBuilder fromXContent(XContentParser parser) throws IOExc
                     "followed by array of terms or a document lookup specification");
         }
 
-        TermsQueryBuilder builder = new TermsQueryBuilder(fieldName, values, termsLookup)
+        return new TermsQueryBuilder(fieldName, values, termsLookup)
             .boost(boost)
             .queryName(queryName);
-
-        if (builder.isTypeless() == false) {
```

Comment:
@romseygeek sorry for the late review, I'm wondering why this check wasn't sufficient for emitting a deprecation warning? Was it not triggered on all the relevant code paths?

It would be nice to keep this strategy if possible, since it's consistent with all our other types deprecation warnings (`deprecatedAndMaybeLog("api_name_with_types", TYPES_DEPRECATION_MESSAGE);`). It also lets us avoid adding an `allowed_warnings` section to the yml tests, since in 7.x we already ignore all warnings in yml tests that start with `[types removal] ... `.

Reply:
There was a path that didn't trigger the warning, but I can't find it now, so it may have been refactored away elsewhere.  I'll have a look, it would be good to remove some of these extra `allowed_warnings` messages.